internal/bot/analyze.go

```diff
@@ -0,0 +1,100 @@
+package bot
+
+import (
+	"context"
+	"fmt"
+	"github.com/austinvalle/hammy/internal/command"
+	"github.com/austinvalle/hammy/internal/llm"
+	"github.com/bwmarrin/discordgo"
+	"github.com/spf13/viper"
+	"golang.org/x/exp/slices"
+	"log/slog"
+	"regexp"
+	"strings"
+)
+
+const (
+	urlPattern = ".*(?P<url>https?:\\/\\/(www\\.)?[-a-zA-Z0-9@:%._\\+~#=]{1,256}\\.[a-zA-Z0-9()]{1,6}\\b([-a-zA-Z0-9()@:%_\\+.~#?&//=]*)).*"
+)
+
+type summarizeCommand struct {
+	logger *slog.Logger
+	llm    *llm.LLM
+}
+
+func newSummarizeCommand(logger *slog.Logger, llm *llm.LLM) command.TextCommand {
+	return &summarizeCommand{
+		logger: logger,
+		llm:    llm,
+	}
+}
+func (c *summarizeCommand) Name() string {
+	return "analyze"
+}
+
+func (c *summarizeCommand) CanActivate(s *discordgo.Session, m discordgo.Message) bool {
+	urlRegex := regexp.MustCompile(urlPattern)
+
+	//don't respond if we are not tagged, embeds don't use mention for bots, its mention role
+	lookupMentionRole := func(roleId string) bool {
+		role, err := s.State.Role(m.GuildID, roleId)
+		if err != nil {
+			c.logger.Error("Unable to lookup discord role for analysis activation", "roleId", m.MentionRoles[0])
+			return false
+		}
+		return role.Name == s.State.User.Username
+	}
+
+	if !slices.ContainsFunc(m.MentionRoles, lookupMentionRole) {
+		return false
+	}
+
+	matches := urlRegex.FindStringSubmatch(m.Content)
+	idx := urlRegex.SubexpIndex("url")
+
+	if len(matches) < idx+1 || matches[idx] == "" {
+		return false
+	}
+
+	return true
+}
+
+func (c *summarizeCommand) Handler(ctx context.Context, s *discordgo.Session, m *discordgo.MessageCreate) (*discordgo.MessageSend, error) {
+	//todo: dont rawdog viper
+	if err := s.MessageReactionAdd(m.ChannelID, m.ID, viper.GetString("ResponseEmoji")); err != nil {
+		c.logger.Error("error adding reaction: ", err)
+	}
+
+	urlRegex := regexp.MustCompile(urlPattern)
+	c.logger.Debug("In summary handler")
+
+	blacklistedSites := []string{"twitter.com", "x.com", "facebook.com", "instagram.com", "reddit.com", "google.com"}
```

Comment:
I was not sure how it would respond with social media comments, I just pull all the text off the page so it gets what it gets and it can slow it down a lot with useless content. (side note if the input is too long we can lose our system instructions so I cut the back part of the content when it is too long, assuming the good stuff is in the front), 

I did try and turn on reddit but they catch headless right away and block the request so you pay for their api instead.